src/modal.rs

```diff
@@ -27,7 +25,7 @@ pub fn find_modal_text(
 
         if text.custom_id == custom_id {
             let value = std::mem::take(&mut text.value);
-            return if value.is_empty() { None } else { Some(value) };
+            return value.and_then(|val| (!val.is_empty()).then_some(val));
```

Comment:
:raised_eyebrow: is this really better?

Reply:
No, I don't know what I was thinking lol, lemme revert that